spec.bs

```diff
@@ -659,7 +659,42 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
 <div algorithm>
   The <dfn for=Observable method><code>map(|mapper|)</code></dfn> method steps are:
 
-    1. <span class=XXX>TODO: Spec this and use |mapper|.</span>
+    1. Let |sourceObservable| be [=this=].
+
+    1. Let |observable| be a [=new=] {{Observable}} whose [=Observable/subscribe callback=] is an
+       algorithm that takes a {{Subscriber}} |subscriber| and does the following:
+
+       1. Let |controller| be a [=new=] {{AbortController}}.
+
+       1. Let |sourceObserver| be a new [=internal observer=], initialized as follows:
+
+          : [=internal observer/next steps=]
+          :: 1. [=Invoke=] |mapper| with the passed in <var ignore>value</var>, and let
```

Comment:
It should also supply an "index" (which is obviously a zero based counter). It doesn't need to supply a `source` argument like array does, because it really doesn't make sense for observable.

But `flatMap`, `map`, `filter`, `scan`, (or anything where there's a handler that takes a value, really) should probably be supplied an `index` that starts at 0 and increments for each value emitted.

Reply:
Done!